flocker/provision/_aws.py

```diff
@@ -67,7 +79,13 @@ def _wait_until_running(instance):
                 instance_state=instance.state,
             ):
                 sleep(1)
-            instance.update()
+            try:
+                instance.update()
+            except EC2ResponseError as e:
+                _check_response_error(
+                    e,
+                    u"flocker:provision:aws:wait_until_running:retry"
+                )
```

Comment:
Likewise - `flocker.common.poll_until` with a timeout might be a good thing to use here.